src/main/java/org/kiwiproject/dropwizard/util/health/ServicePingHealthCheck.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * Registers a {@link ServicePingHealthCheck} for each of the given {@link ServiceIdentifier}s given, defaulting the
+     * importance to REQUIRED.
+     *
+     * @param healthCheckRegistry   the {@link HealthCheckRegistry} to register the health checks.
+     * @param client                the {@link RegistryAwareClient} to use for service lookup and to perform the request.
+     * @param serviceIdentifiers    the identifiers for the service to be pinged.
+     */
+    public static void registerServiceChecks(HealthCheckRegistry healthCheckRegistry,
+                                             RegistryAwareClient client,
+                                             ServiceIdentifier... serviceIdentifiers) {
+
+        registerServiceChecks(healthCheckRegistry, client, DependencyImportance.REQUIRED, serviceIdentifiers);
+    }
+
+    /**
+     * Registers a {@link ServicePingHealthCheck} for each of the given {@link ServiceIdentifier}s given, using the
+     * given importance <em>for each</em>.
+     * <p>
+     * NOTE: If different importances are required for different services, then call this method for each importance.
+     *
+     * @param healthCheckRegistry   the {@link HealthCheckRegistry} to register the health checks.
+     * @param client                the {@link RegistryAwareClient} to use for service lookup and to perform the request.
+     * @param importance            the {@link DependencyImportance} of the services being checked.
+     * @param serviceIdentifiers    the identifiers for the service to be pinged.
+     */
+    public static void registerServiceChecks(HealthCheckRegistry healthCheckRegistry,
+                                             RegistryAwareClient client,
+                                             DependencyImportance importance,
+                                             ServiceIdentifier... serviceIdentifiers) {
+
+        checkArgumentNotNull(healthCheckRegistry, "healthCheckRegistry is required");
+        checkArgumentNotNull(client, "client is required");
+
+        if (isNullOrEmpty(serviceIdentifiers)) {
+            LOG.warn("No services were supplied, so no ServicePingHealthChecks will be added");
+            return;
+        }
+
+        Stream.of(serviceIdentifiers).forEach(identifier -> {
+            var healthCheckName = nameFor(identifier);
+            var healthCheck = new ServicePingHealthCheck(identifier, client, importance);
+            healthCheckRegistry.register(healthCheckName, healthCheck);
+        });
+    }
+
+    /**
+     * Resolves a health check name for a given {@link ServiceIdentifier}.
+     *
+     * @param identifier The {@link ServiceIdentifier} with a name being resolved.
+     * @return a health check name to match the given {@link ServiceIdentifier}.
+     */
+    public static String nameFor(ServiceIdentifier identifier) {
+        var humanizedName = WordUtils.capitalizeFully(identifier.getServiceName(), '-')
```

Comment:
This will require `commons-text` to be on the class path, and the POM doesn't explicitly require it, which means its coming in transitively. Do you know where it is coming from?

BTW I was wrong earlier...the Guava utility for making conversions like this is `CaseFormat`: https://github.com/google/guava/wiki/StringsExplained#caseformat

But it doesn't seem to provide a way to humanize, though you can do:

```java
CaseFormat.LOWER_HYPHEN.to(CaseFormat.UPPER_CAMEL, "test-service")

// TestService
```

which isn't what you want here, unfortunately. But if it's good enough, then we could use it.

Or we could just use the service name "as is", i.e. just accept `test-service`

Why these things are not in the JDK directly, I do not understand. Maybe only b/c it's specific to certain languages?

Reply:
It comes from dropwizard-core --> dropwizard-configurations